src/main/java/com/exadel/discountwebapp/validation/CategoryValidator.java

```diff
@@ -15,6 +17,7 @@ public void validate(CategoryRequestVO request) {
         checkDuplicateTitle(request);
     }
 
+    @Transactional(propagation = Propagation.REQUIRED, readOnly = true)
     public void checkDuplicateTitle(CategoryRequestVO request) {
         categoryRepository.findByTitle(request.getTitle())
                 .ifPresent(category -> {
```

Comment:
instead you can use categoryRepository.existsById

Reply:
changed to existsByTitle() and added such method to repository